docs/end_to_end_tutorials/chatbots/building_a_chatbot.md

```diff
@@ -79,143 +92,70 @@ for year in years:
     index_set[year] = cur_index
 ```
 
+### Setting up a Sub Question Query Engine to Synthesize Answers Across 10-K Filings
 
-### Composing a Graph to Synthesize Answers Across 10-K Filings
-
-Since we have access to documents of 4 years, we may not only want to ask questions regarding the 10-K document of a given year, but ask questions that require analysis over all 10-K filings. 
+Since we have access to documents of 4 years, we may not only want to ask questions regarding the 10-K document of a given year, but ask questions that require analysis over all 10-K filings.
 
-To address this, we compose a "graph" which consists of a summary index defined over the 4 vector indices. Querying this graph would first retrieve information from each vector index, and combine information together via the summary index.
+To address this, we can use a [Sub Question Query Engine](https://gpt-index.readthedocs.io/en/stable/examples/query_engine/sub_question_query_engine.html). It decomposes a query into subqueries, each answered by an individual vector index, and synthesizes the results to answer the overall query.
 
 ```python
-from llama_index import SummaryIndex, LLMPredictor, ServiceContext, load_graph_from_storage
-from llama_index.llms import OpenAI
-from llama_index.indices.composability import ComposableGraph
-
-# describe each index to help traversal of composed graph
-index_summaries = [f"UBER 10-k Filing for {year} fiscal year" for year in years]
-
-# define an LLMPredictor set number of output tokens
-llm = OpenAI(temperature=0, max_tokens=512, model="gpt-4")
-service_context = ServiceContext.from_defaults(llm=llm)
-storage_context = StorageContext.from_defaults()
-
-# define a summary index over the vector indices
-# allows us to synthesize information across each index
-graph = ComposableGraph.from_indices(
-    SummaryIndex,
-    [index_set[y] for y in years], 
-    index_summaries=index_summaries,
-    service_context=service_context,
-    storage_context = storage_context,
-)
-root_id = graph.root_id
-
-# [optional] save to disk
-storage_context.persist(persist_dir=f'./storage/root')
-
-# [optional] load from disk, so you don't need to build graph from scratch
-graph = load_graph_from_storage(
-    root_id=root_id, 
-    service_context=service_context,
-    storage_context=storage_context,
-)
-
+from llama_index.tools import QueryEngineTool, ToolMetadata
+from llama_index.query_engine import SubQuestionQueryEngine
 ```
 
-### Setting up the Tools + Langchain Chatbot Agent
-
-We use Langchain to setup the outer chatbot agent, which has access to a set of Tools.
-LlamaIndex provides some wrappers around indices and graphs so that they can be easily used within a Tool interface.
+LlamaIndex provides some wrappers around indices (and query engines) so that they can be used by query engines and agents. First we define a `QueryEngineTool` for each vector index.
```

Comment:
Lets add a quick explanation for the names/descriptions -- this is what the LLM sees when deciding which indexes to use